src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.idl

```diff
@@ -5,20 +5,27 @@ import "ColorSchemeViewModel.idl";
 
 namespace Microsoft.Terminal.Settings.Editor
 {
+    enum ColorSchemesSubPage
+    {
+        Base = 0,
+        EditColorScheme = 1
+    };
+
     runtimeclass ColorSchemesPageViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
     {
         ColorSchemesPageViewModel(Microsoft.Terminal.Settings.Model.CascadiaSettings settings);
         void UpdateSettings(Microsoft.Terminal.Settings.Model.CascadiaSettings settings);
 
-        void AddNew_Click(IInspectable sender, Windows.UI.Xaml.RoutedEventArgs e);
-
-        void RequestEnterRename();
-        Boolean RequestExitRename(Boolean saveChanges, String newName);
+        ColorSchemeViewModel RequestAddNew();
+        Boolean RequestRenameCurrentScheme(String newName);
         void RequestDeleteCurrentScheme();
 
         ColorSchemeViewModel CurrentScheme;
+        Boolean HasCurrentScheme { get; };
         Boolean CanDeleteCurrentScheme { get; };
-        Boolean InRenameMode { get; };
+        ColorSchemesSubPage CurrentPage;
         Windows.Foundation.Collections.IObservableVector<ColorSchemeViewModel> AllColorSchemes { get; };
+
+        void Edit_Click(IInspectable sender, Windows.UI.Xaml.RoutedEventArgs e);

```

Comment:
this should not need to be in the IDL.

If it's in the IDL, _anybody could call it_ even, again, from another DLL. 
That's not a great public API :)